pyproject.toml

```diff
@@ -91,6 +91,7 @@ youtube-transcript-api = "^0.6.1"
 beautifulsoup4 = "^4.12.2"
 pypdf = "^3.11.0"
 pytube = "^15.0.0"
+google-generativeai = "^0.1.0"
```

Comment:
we should make this an extra dependency.

Reply:
Hey, you completely removed it from `pyproject.toml`. I meant to make it optional, and add it under `[tool.poetry.extras]`. In the class, add something similar to:

```python
        try:
            from gpt4all import GPT4All
        except ModuleNotFoundError:
            raise ModuleNotFoundError(
                "The GPT4All python package is not installed. Please install it with `pip install --upgrade embedchain[opensource]`"  # noqa E501
            ) from None

```

Reply:
Dependency looks good now. Thanks!